ENH - Add new sidebar design

steff456 opened the pull request:
Fixes #283.
<!-- Reference the issue corresponding to this PR. If an issues does not exist, consider opening one or writing a detailed description for what this PR changes and it's value in the following sections. -->
<!-- GitHub Docs on Keywords: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests -->

## Description
<!-- What is the purpose of this pull request? -->

This pull request:

- Implements the new sidebar design

**After**

<img width="959" alt="image" src="https://github.com/conda-incubator/conda-store-ui/assets/20992645/0e5455c2-4f89-4307-b152-f67c38c4db03">


## Pull request checklist
<!-- Quick checklist to ensure high-quality Pull Request. -->

- [x] Did you test this change locally?
- [x] Did you update the documentation (if required)?
- [x] Did you add/update relevant tests for this change (if required)?

## Additional information
<!-- Do you have any other information about this pull request? This may include screenshots, references, and/or implementation notes. -->

There's some differences between the given design and the changes of this PR because the implementation may require a complete re-write of whole components, @smeragoel please let me know if there's another way to handle each situation or if the change is required as the figma design you shared with me.

1. The horizontal gray lines don't have the same width

**Current state**
<img width="353" alt="image" src="https://github.com/conda-incubator/conda-store-ui/assets/20992645/1bed1dce-a417-4948-a05c-15794e70fb49">

This happens because all the margins for the sidebar are defined in the most outer box, 

<img width="359" alt="image" src="https://github.com/conda-incubator/conda-store-ui/assets/20992645/cd425a26-01db-4baa-bf13-021ad3b27817">

So the line in of the bottom will always look shorter than the top line

2. The active environment green background only highlights the text, without the circle icon

**Current state**
<img width="191" alt="image" src="https://github.com/conda-incubator/conda-store-ui/assets/20992645/3b7b4095-2f17-4b44-8c87-6b8d7e0eb6c7">

**Design**
<img width="222" alt="image" src="https://github.com/conda-incubator/conda-store-ui/assets/20992645/430aab03-1255-41d4-b5fa-8f67806d303a">

3. Do we want to use a personalized icon, or the mui icon that accompanies the `Shared Environment` title?

**Current state**
<img width="261" alt="image" src="https://github.com/conda-incubator/conda-store-ui/assets/20992645/799bb293-0d6e-4a82-ae8a-51b5abd9b8bd">

**Design**
<img width="200" alt="image" src="https://github.com/conda-incubator/conda-store-ui/assets/20992645/67e14129-3351-4cfa-ba11-7119919057ae">

smeragoel commented:
1. For the width of the horizontal lines, I'd prefer if they are the same width because it looks a little wonky if they are not. If it requires significant changes to the code, we can also remove the bottom line to resolve this quickly.
2. That's completely fine, in fact, this is what I originally intended to do!
3. How hard will it be to use a personalised icon, not just here but in other places as well? I am aiming to use the iconset we have defined in the design system. That will also require changes across other places we have used icons eventually.

smeragoel commented:
Here's the SVG:
![Group_light](https://github.com/conda-incubator/conda-store-ui/assets/98317216/0440a800-8f1e-42da-ae30-798bc115a4b3)

You can also export it directly from Figma if you need it in a different size: [Group Icon](https://www.figma.com/file/nKifk755TFMsK2SF082igS/conda-store-UI-design?type=design&node-id=1461%3A1066&mode=design&t=u0OtvGTOU8XmM4cq-1)
([Here's](https://help.figma.com/hc/en-us/articles/360040028114-Export-from-Figma) some documentation on the export tool in Figma, in case you need it!)

steff456 commented:
@smeragoel this PR is ready for another review! I fixed the items discussed in our conversation and also fixed the typography that was off in the `read the docs!` link. Please let me know if there's anything else that needs to be addressed or if this PR is ready to merge!
